src/assets/wise5/classroomMonitor/classroomMonitorComponents/nodeGrading/nodeGradingView/nodeGradingView.ts

```diff
@@ -196,113 +186,81 @@ export class NodeGradingViewController {
    * @param workgroupID a workgroup ID number
    * @param init Boolean whether we're in controller initialization or not
    */
-  updateWorkgroup(workgroupId, init = false) {
+  updateWorkgroup(workgroupId: number, init = false) {
     const workgroup = this.workgroupsById[workgroupId];
-    if (workgroup) {
-      const alertNotifications = this.getAlertNotificationsByWorkgroupId(workgroupId);
-      workgroup.hasAlert = alertNotifications.length;
-      workgroup.hasNewAlert = this.workgroupHasNewAlert(alertNotifications);
-      const completionStatus = this.getCompletionStatusByWorkgroupId(workgroupId);
-      //workgroup.hasNewWork = completionStatus.hasNewWork;
-      workgroup.isVisible = completionStatus.isVisible ? 1 : 0;
-      workgroup.completionStatus = this.getWorkgroupCompletionStatus(completionStatus);
-      workgroup.score = this.getScoreByWorkgroupId(workgroupId);
-      if (!init) {
-        this.workgroupsById[workgroupId] = angular.copy(workgroup);
-      }
-    }
-  }
-
-  getAlertNotificationsByWorkgroupId(workgroupId) {
-    const args = {
+    const alertNotifications = this.NotificationService.getAlertNotifications({
       nodeId: this.nodeId,
       toWorkgroupId: workgroupId
-    };
-    return this.NotificationService.getAlertNotifications(args);
+    });
+    workgroup.hasAlert = alertNotifications.length > 0;
+    workgroup.hasNewAlert = this.workgroupHasNewAlert(alertNotifications);
+    const completionStatus = this.getCompletionStatusByWorkgroupId(workgroupId);
+    workgroup.isVisible = completionStatus.isVisible ? 1 : 0;
+    workgroup.completionStatus = this.getWorkgroupCompletionStatus(completionStatus);
+    workgroup.score = this.getScoreByWorkgroupId(workgroupId);
+    if (!init) {
+      this.workgroupsById[workgroupId] = angular.copy(workgroup);
+    }
   }
 
-  workgroupHasNewAlert(alertNotifications) {
-    let newAlert = false;
-    let l = alertNotifications.length;
-    for (let i = 0; i < l; i++) {
-      let alert = alertNotifications[i];
+  workgroupHasNewAlert(alertNotifications: Notification[]): boolean {
+    for (const alert of alertNotifications) {
       if (!alert.timeDismissed) {
-        newAlert = true;
-        break;
+        return true;
       }
     }
-
-    return newAlert;
+    return false;
   }
 
-  getCompletionStatusByWorkgroupId(workgroupId) {
-    let isCompleted = false;
-    let isVisible = false;
-    let latestWorkTime = null;
-    let latestAnnotationTime = null;
+  getCompletionStatusByWorkgroupId(workgroupId: number): CompletionStatus {
+    const status: CompletionStatus = {
```

Comment:
Should this status variable name be changed to completionStatus? In this function there is a studentStatus variable and a nodeStatus variable and if you have another variable just called status it's a little confusing.

Reply:
good point, I changed it to completionStatus.